ukaea/piezo
# Feature/validation rules as a config map #100

Load validation rules as a configmap instead of by building them into the docker image. This allows for the validation rules to be updated without needing the docker image to be rebuilt. Note this only applies when the web app is running in a pod. For development purposes of running the web app locally the validation rules are read directly from the files. 
Instruction on the steps required to create the configmap and run the web app using it have been added to the deployment guide in the wiki: https://github.com/ukaea/piezo/wiki/WebAppDeployment#validation-rules

Also added is an automated script to run after updating the validation rules file. This will restart the web app using the new data without needing to rebuild the docker image.  The reasoning for using the chosen method here have been documented in the ADR https://github.com/ukaea/piezo/wiki/WebAppDecisionRecord#updating-validation-rules  

## Stories covered
#100 

## To test
* Deploy the web app in a pod using a configmap
* Try running a job with an invalid argument (executors = 11)
* It should fail
* Edit validation rules to make argument valid (max executors = 12)
* Run update_validation_rules.sh
* Try rerunning the same job
* It should now pass
